src/lib.rs

```diff
@@ -727,20 +730,31 @@ impl Client {
     /// future. In particular, in time, it may only run `is_ready` again when an event occurs on
     /// the page.
     pub async fn wait_for_find(&mut self, search: Locator<'_>) -> Result<Element, error::CmdError> {
-        let s: webdriver::command::LocatorParameters = search.into();
-        loop {
-            match self
-                .by(webdriver::command::LocatorParameters {
-                    using: s.using,
-                    value: s.value.clone(),
-                })
-                .await
-            {
-                Ok(v) => break Ok(v),
-                Err(error::CmdError::NoSuchElement(_)) => {}
-                Err(e) => break Err(e),
+        fn closure(
+            s: Locator<'_>,
+            mut client: Client, // TODO: handle lifetime issues with &mut Client type
+        ) -> impl Future<Output = Option<Result<Element, error::CmdError>>> {
+            let s: webdriver::command::LocatorParameters = s.into();
+            async move {
+                match client
+                    .by(webdriver::command::LocatorParameters {
+                        using: s.using,
+                        value: s.value.clone(),
+                    })
+                    .await
+                {
+                    Ok(v) => Some(Ok(v)),
+                    Err(error::CmdError::NoSuchElement(_)) => None,
+                    Err(e) => Some(Err(e)),
+                }
             }
         }
+
+        const DEFAULT_TIMEOUT: std::time::Duration = std::time::Duration::from_millis(500);
```

Comment:
I think, by default, this method should never time out. The user asked us to wait indefinitely. If the user wants a timeout, they can do so by wrapping the returned future in a `tokio::time::timeout`.

Reply:
I hope that you're speaking about changes in `wait_for` here's you are right.
It's just a method which I chose to demonstrate the `Wait` :smile: and didn't noticed that I break it's logic :smile: 

Probably there could be introduced a new type WaitForever which would do just `loop`. And after we could amalgamate these by a trait `Waiter` which has a method `until`.

Reply:
My argument was more that all you need is the ability to wait forever. If the user wants to stop waiting before the action succeeds, they can just wrap the future in `tokio::time::timeout` :)